service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportItkHeaderParserUtil.java

```diff
@@ -23,8 +24,22 @@ public ItkReportHeader getHeaderValues(Element headerElement) {
 
     private void setAddressList(ItkReportHeader header, Element headerElement) {
         List<Element> elements = getElements(headerElement, ITK_ADDRESS_NODE);
-        List<String> addresses = elements.stream().map(it -> it.attributeValue("uri")).collect(toList());
-        header.setAddressList(addresses);
+        Optional<String> dosServiceId = elements.stream()
+            .filter(it -> it.attribute("type") != null)
+            .map(it -> it.attributeValue("uri"))
+            .findFirst();
+
+        List<String> addressList = elements.stream()
+            .filter(it -> it.attribute("type") == null
+                && it.attributeValue("uri").contains("ods")
+            )
+            .map(it -> dosServiceId.isPresent()
+                ? it.attributeValue("uri") + ":DOSServiceID:" + dosServiceId.get()
+                : it.attributeValue("uri")
+            )
+            .collect(toList());
```

Comment:
I'm not sure if ITK message can contain DOS ID only (no ODS code). This code assumes ODS is always there, I think we should treat it as optional.

Reply:
Changed to allow for ODS + DOS ID, only ODS and only DOS ID.